api/versionedproposal.go

```diff
@@ -0,0 +1,52 @@
+// Copyright © 2023 Attestant Limited.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	apiv1deneb "github.com/attestantio/go-eth2-client/api/v1/deneb"
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/capella"
+)
+
+// VersionedProposal contains a versioned beacon node proposal.
+type VersionedProposal struct {
+	Version   spec.DataVersion
+	Bellatrix *bellatrix.BeaconBlock
+	Capella   *capella.BeaconBlock
+	Deneb     *apiv1deneb.BlockContents
+}
+
+// String returns a string version of the structure.
+func (v *VersionedProposal) String() string {
+	switch v.Version {
+	case spec.DataVersionBellatrix:
```

Comment:
Why are we not checking for phase0 blocks?

Reply:
This was the result of a partial implementation during the move to using proposals.  Should now be fully updated with all forks.